libraries/AP_HAL_SITL/system.cpp

```diff
@@ -96,6 +96,13 @@ static void run_command_on_ownpid(const char *commandname)
         p = progname;
     }
 
+    // work around a bug in dash; its manpage says it searches $PATH
+    // but it does't.  https://github.com/ArduPilot/ardupilot/issues/24245
+    const char *cmd_fmt = "sh %s %d >%s 2>&1";
+    if (getenv("AP_DASH_HACKERY")) {
```

Comment:
how does this env variable get set?

Reply:
By the user.

They already need to be running SITL (or autotest) with their cwd set to a place we're not expecting, and to have modified their $PATH to include `scripts`, so I figured an additional environment variable would be OK.

This patch was designed as least-risk-to-existing-users.

Reply:
I'm thinking it might be easier to understand (for the users like me w/ weird setups) if the env var was something more like AP_OVERRIDE_PATH_TO_SCRIPTS, to just specify the full path to where dumpstack/dumpcore are to be found, rather than the two-step process of adding that dir to PATH and also setting this other var.

Reply:
> I'm thinking it might be easier to understand (for the users like me w/ weird setups) if the env var was something more like AP_OVERRIDE_PATH_TO_SCRIPTS=/path/to/scripts, to just specify the full path to where dumpstack/dumpcore are to be found, rather than the two-step process of adding that dir to PATH and also setting this other var.

Yeah, I considered doing this approach but went with your suggestion instead as a tested solution :-)

Decided at DevCall to go with this, looking for an env variable speciying the path.  The reason I prefer this one is that you don't need to modify your path to find the tool....